spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveViews.scala

```diff
@@ -59,6 +61,32 @@ case class ResolveViews(spark: SparkSession) extends Rule[LogicalPlan] with Look
       loadView(catalog, ident)
         .map(_ => ResolvedV2View(catalog.asViewCatalog, ident))
         .getOrElse(u)
+
+    case c@CreateIcebergView(ResolvedIdentifier(_, ident), _, query, columnAliases, columnComments, _, _, _, _, _,
+    rewritten)
```

Comment:
Rather than using an extra `rewritten` flag, you can override `resolved` in `CreateIcebergView`. Then this rule can look for unresolved `CreateIcebergView` instances where `query.resolved` is true.

Reply:
I tried that but `resolved` is defined as `lazy val resolved: Boolean` so we'd have to lazily compute its value based on a different flag, hence it seemed easier to just have the `rewritten` flag